src/main/java/com/team4/testingsystem/services/impl/FileAnswerServiceImpl.java

```diff
@@ -49,10 +49,22 @@ public String getUrl(Long testId, Long questionId) {
     }
 
     @Override
-    public FileAnswer uploadSpeaking(MultipartFile file, Long testId, Modules module) {
+    public FileAnswer uploadSpeaking(MultipartFile file, Long testId) {
+        Test test = testsService.getById(testId);
+
+        testsService.checkRights(test);
+
+        testsService.checkStartedStatus(test);
+
         String url = storageService.upload(file.getResource(), Modules.SPEAKING, testId);
-        Question question = questionService.getQuestionByTestIdAndModule(testId, module);
-        return save(testId, question.getId(), url);
+
+        Question question = questionService.getQuestionByTestIdAndModule(testId, Modules.SPEAKING);
+
+        FileAnswer fileAnswer = FileAnswer.builder()
+            .id(new TestQuestionID(test, question))
+            .url(url)
+            .build();
+        return fileAnswerRepository.save(fileAnswer);
```

Comment:
Will it also work properly with update?

Reply:
Why not?
We rewrite file answer if it already exists

Reply:
We expected the behavior, but seams like something is broken there